### PR TITLE
[PWD-2569] Adjust Register for Corporate Sale Start and End Date

### PR DESCRIPTION
**Pull request by haikal1223**
…ales date

### What type of PR is this? (_check all applicable_)

- [x] Feature
- [ ] Bug Fix
- [ ] Documentation Update
- [ ] Setup
- [ ] Code Refactor
- [ ] Performance Improvements
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Revert

### CODE REVIEW

File: src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx [hunk 1]
```diff
@@ -35,6 +42,38 @@ const IntakeCard: FC<IntakeCardProps> = ({
   const desktop = useMediaQuery(theme.breakpoints.down('lg'))
 
   const { days, hours } = practiceSessionSchedule
+  const { corporateSaleStartDate, corporateSaleEndDate } = intakeItem
+  const currentDate = new Date()
+  const convertCorporateSaleStartDate = new Date(corporateSaleStartDate)
+  const convertCorporateSaleEndDate = new Date(corporateSaleEndDate)
+
+  const checkRegisterRequirement = useMemo(() => {
```
Comment (shadiqfauzi) [inline src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx:50]:
```suggestion
  const checkRegisterRequirement = useMemo(() => {
```
nama variablenya ga cocok pake "check" karena returnnya boolean. lebih cocok "is" atau yg mirip istlahnya. case disini bisa kaya misalnya "isNotAvailableForSale"

File: src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx [hunk 2]
```diff
@@ -35,6 +42,38 @@ const IntakeCard: FC<IntakeCardProps> = ({
...
+    )
+  }, [intakeItem.corporatePrice, intakeItem])
+
+  const toolTipMessage = useMemo(() => {
+    if (
+      intakeItem.corporatePrice === 0 &&
+      (currentDate < convertCorporateSaleStartDate ||
+        currentDate > convertCorporateSaleEndDate)
+    ) {
+      return 'price & date not available, please contact administrator'
```
Comment (shadiqfauzi) [inline src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx:64]:
```suggestion
      return 'price & sale date not available, please contact administrator'
```

File: src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx [hunk 3]
```diff
@@ -35,6 +42,38 @@ const IntakeCard: FC<IntakeCardProps> = ({
...
+    }
+    if (intakeItem.corporatePrice === 0) {
+      return 'price not available, please contact administrator'
+    }
+
+    if (
+      currentDate < convertCorporateSaleStartDate ||
+      currentDate > convertCorporateSaleEndDate
+    ) {
+      return 'date not available, please contact administrator'
```
Comment (shadiqfauzi) [inline src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx:74]:
```suggestion
      return 'sale date not available, please contact administrator'
```

File: src/views/register-program/components/Content/index.tsx [hunk 4]
```diff
@@ -27,14 +27,22 @@ const Content: FC<ContentProps> = ({
...
+  const totalPrice = useTotalPrice(formik, price)
 
   const disabledButton = useMemo<boolean>(() => {
     const totalPayment = voucher - totalPrice
-    const zeroPriceProgram = type === 'program' && data.price === 0
+    const programDisabled =
+      type === 'program' &&
+      (price === 0 ||
+        currentDate < convertCorporateSaleStartDate ||
+        currentDate > convertCorporateSaleEndDate)
```
Comment (shadiqfauzi) [inline src/views/register-program/components/Content/index.tsx:43]: pusing bacanya. dipisah ke variable aja kalau udah panjang gini condition

File: src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx [hunk 5]
```diff
@@ -35,6 +42,38 @@ const IntakeCard: FC<IntakeCardProps> = ({
...
+      currentDate < convertCorporateSaleStartDate ||
+      currentDate > convertCorporateSaleEndDate
+    )
+  }, [intakeItem.corporatePrice, intakeItem])
+
+  const toolTipMessage = useMemo(() => {
+    if (
+      intakeItem.corporatePrice === 0 &&
+      (currentDate < convertCorporateSaleStartDate ||
+        currentDate > convertCorporateSaleEndDate)
```
Comment (shadiqfauzi) [inline src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx:62]: ini juga

File: src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx (hunk 1 above)
Comment (haikal1223) [reply src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx:50]: hooo okehh mas

File: src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx [hunk 6]
```diff
@@ -35,6 +42,34 @@ const IntakeCard: FC<IntakeCardProps> = ({
   const desktop = useMediaQuery(theme.breakpoints.down('lg'))
 
   const { days, hours } = practiceSessionSchedule
+  const { corporateSaleStartDate, corporateSaleEndDate } = intakeItem
+  const currentDate = new Date()
+  const convertCorporateSaleStartDate = new Date(corporateSaleStartDate)
+  const convertCorporateSaleEndDate = new Date(corporateSaleEndDate)
+
+  const isCorporateDateRange = useMemo(() => {
```
Comment (shadiqfauzi) [inline src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx:50]:
```suggestion
  const isInCorporateSaleDateRange = useMemo(() => {
```

File: src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx [hunk 7]
```diff
@@ -35,6 +42,34 @@ const IntakeCard: FC<IntakeCardProps> = ({
   const desktop = useMediaQuery(theme.breakpoints.down('lg'))
 
   const { days, hours } = practiceSessionSchedule
+  const { corporateSaleStartDate, corporateSaleEndDate } = intakeItem
+  const currentDate = new Date()
+  const convertCorporateSaleStartDate = new Date(corporateSaleStartDate)
+  const convertCorporateSaleEndDate = new Date(corporateSaleEndDate)
```
Comment (shadiqfauzi) [inline src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx:48]: cuma dipake di variable date range kan? pindahin kesitu aja

Comment (haikal1223) [reply src/views/corporate-training/detail/fixed-programs/sections/JadwalKelas/components/IntakeList/IntakeCard.tsx:48]: okehh

**Review by Vesals — APPROVED** [review]

**Review by kevintanes — APPROVED** [review]

**Pull request merged**